lib/executor/builtin.cpp

```diff
@@ -0,0 +1,32 @@
+
+#include "shell/executor/builtins.hpp"
+
+namespace shell
+{
+
+	Builtin& Builtin::getInstance() {
+		static Builtin builtin;
+		return builtin;
+	}
+
+	Builtin::Builtin() {
+		builtins.emplace("exit", [=](const Ast::Command& command) {
+			exit(command.args);
+			return ResultCode::Ok; // Does not actually get hit because it exits, its just here to keep the compiler happy :)
+		});
+
+		builtins.emplace("echo", [=](const Ast::Command& command) {
+			return echo(command.args);
```

Comment:
this can probably be `builtins.emplace("echo", echo);`, no?
If `echo` is a static method matching the prototype for the `builtinFunction`